core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -639,10 +640,11 @@ private[deploy] class Master(
     // in the queue, then the second app, etc.
     for (app <- waitingApps if app.coresLeft > 0) {
       val coresPerExecutor: Option[Int] = app.desc.coresPerExecutor
+      val coresPerTask = app.desc.coresPerTask
       // Filter out workers that don't have enough resources to launch an executor
       val usableWorkers = workers.toArray.filter(_.state == WorkerState.ALIVE)
         .filter(worker => worker.memoryFree >= app.desc.memoryPerExecutorMB &&
-          worker.coresFree >= coresPerExecutor.getOrElse(1))
+          worker.coresFree >= math.max(coresPerExecutor.getOrElse(1), coresPerTask))
```

Comment:
IMO it's not clear what the relationship between coresPerTask and coresPerExecutor. Looking at the scheduling requirement we just need the maximum of the set to be able to schedule, and I thought cores per task is additional cpu resources on top of the executor. 
Can we perhaps comment where this is introduced what these two are? Or point to documentation?